internal/gtfs/direction_calculator.go

```diff
@@ -0,0 +1,131 @@
+package gtfs
+
+import (
+	"context"
+	"maglev.onebusaway.org/gtfsdb"
+	"maglev.onebusaway.org/internal/utils"
+)
+
+type DirectionCalculator struct {
+	queries *gtfsdb.Queries
+}
+
+func NewDirectionCalculator(queries *gtfsdb.Queries) *DirectionCalculator {
+	return &DirectionCalculator{
+		queries: queries,
+	}
+}
+
+// CalculateStopDirection determines the compass direction for a stop
+func (dc *DirectionCalculator) CalculateStopDirection(ctx context.Context, stopID string) string {
+	// Strategy 1: Try shape-based calculation
+	if direction := dc.calculateFromShape(ctx, stopID); direction != "" {
+		return direction
+	}
+
+	// Strategy 2: Fallback to stop-to-stop calculation
+	if direction := dc.calculateFromNextStop(ctx, stopID); direction != "" {
+		return direction
+	}
+
+	// No direction could be calculated
+	return ""
+}
+
+func (dc *DirectionCalculator) calculateFromShape(ctx context.Context, stopID string) string {
+	// Get trips serving this stop
+	stopTrips, err := dc.queries.GetStopsWithTripContext(ctx, stopID)
+	if err != nil || len(stopTrips) == 0 {
+		return ""
```

Comment:
This should probably be `"UNKNOWN"`, as should any other cases where we return early on an error. (e.g. in `calculateFromNextStop()`. Alternatively, these functions that can produce errors should return a string `(string, error)` tuple. I don't actually feel strongly about this one way or another, as long as we're consistent and the API output always reads "UNKNOWN"